### PR TITLE
Add namespace qualifiers to size()

### LINKED ISSUE #1691: namespace qualify uses of `stan::math::size()`.
## Description

There is a new `size()` function in some C++17 compilers (maybe even the standard) that is causing ambiguities with Stan's `size()` function.  All the uses in the Stan codebase of `size()` should be qualified to `stan::math::size()`.

See, e.g., PR #1684, which fixed one instance.

#### Current Version:
v3.1.0

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

Fixes #1691 by adding stan::math:: to all uses of the size function in stan/math/prim/fun/size.hpp.

## Tests

/

## Side Effects

/

Copyright holder: Rok Češnovar

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
I was so focused on not missing any of them and not adding it to all the .size() calls that I put them to the declarations. Fixed now. Should be good to go.

**Comment by bob-carpenter** [comment]
Looks like there are still some cases that didn't get cleaned up.  This will be good to go as soon as tests pass.

**Comment by rok-cesnovar** [comment]
Ok, green now. Ready for approval.

File: stan/math/prim/prob/beta_log.hpp [hunk 1]
```diff
@@ -21,8 +21,8 @@ namespace math {
  * @deprecated use <code>beta_lpdf</code>
  *
  * @param y (Sequence of) scalar(s).
- * @param alpha (Sequence of) prior sample size(s).
- * @param beta (Sequence of) prior sample size(s).
+ * @param alpha (Sequence of) prior sample stan::math::size(s).
+ * @param beta (Sequence of) prior sample stan::math::size(s).
```
Comment (mcol) [inline stan/math/prim/prob/beta_log.hpp]: Could you change this back? There's an other case before in the same file.

File: stan/math/prim/prob/beta_lpdf.hpp [hunk 2]
```diff
@@ -19,17 +19,17 @@ namespace math {
...
  * <p> The result log probability is defined to be the sum of
  * the log probabilities for each observation/alpha/beta triple.
  *
  * Prior sample sizes, alpha and beta, must be greater than 0.
  *
  * @param y (Sequence of) scalar(s).
- * @param alpha (Sequence of) prior sample size(s).
- * @param beta (Sequence of) prior sample size(s).
+ * @param alpha (Sequence of) prior sample stan::math::size(s).
+ * @param beta (Sequence of) prior sample stan::math::size(s).
```
Comment (mcol) [inline stan/math/prim/prob/beta_lpdf.hpp]: Could you change this back? There's an other case before in the same file.

File: stan/math/prim/prob/normal_sufficient_lpdf.hpp [hunk 3]
```diff
@@ -32,7 +32,7 @@ namespace math {
  * @tparam T_scale Type of scale parameter.
  * @param y_bar (Sequence of) scalar(s) (sample average(s)).
  * @param s_squared (Sequence of) sum(s) of sample squared errors
- * @param n_obs (Sequence of) sample size(s)
+ * @param n_obs (Sequence of) sample stan::math::size(s)
```
Comment (mcol) [inline stan/math/prim/prob/normal_sufficient_lpdf.hpp]: Could you change this back?

**Comment by rok-cesnovar** [comment]
Done, thanks @mcol!

**Comment by rok-cesnovar** [comment]
@bob-carpenter tests are passing now. Ready for another look / approval

**Pull request merged**